src/workspace-element.js

```diff
@@ -222,21 +222,101 @@ class WorkspaceElement extends HTMLElement {
     this.model.getActivePane().activate()
   }
 
-  focusPaneViewAbove () { this.paneContainer.focusPaneViewAbove() }
+  focusPaneViewAbove () { this.focusPaneViewInDirection('above') }
 
-  focusPaneViewBelow () { this.paneContainer.focusPaneViewBelow() }
+  focusPaneViewBelow () { this.focusPaneViewInDirection('below') }
 
-  focusPaneViewOnLeft () { this.paneContainer.focusPaneViewOnLeft() }
+  focusPaneViewOnLeft () { this.focusPaneViewInDirection('left') }
 
-  focusPaneViewOnRight () { this.paneContainer.focusPaneViewOnRight() }
+  focusPaneViewOnRight () { this.focusPaneViewInDirection('right') }
 
-  moveActiveItemToPaneAbove (params) { this.paneContainer.moveActiveItemToPaneAbove(params) }
+  focusPaneViewInDirection (direction, pane) {
+    const activePane = this.model.getActivePane()
+    const paneToFocus = this.nearestVisiblePaneInDirection(direction, activePane)
+    paneToFocus && paneToFocus.focus()
+  }
+
+  moveActiveItemToPaneAbove (params) {
+    this.moveActiveItemToNearestPaneInDirection('above', params)
+  }
 
-  moveActiveItemToPaneBelow (params) { this.paneContainer.moveActiveItemToPaneBelow(params) }
+  moveActiveItemToPaneBelow (params) {
+    this.moveActiveItemToNearestPaneInDirection('below', params)
+  }
 
-  moveActiveItemToPaneOnLeft (params) { this.paneContainer.moveActiveItemToPaneOnLeft(params) }
+  moveActiveItemToPaneOnLeft (params) {
+    this.moveActiveItemToNearestPaneInDirection('left', params)
+  }
+
+  moveActiveItemToPaneOnRight (params) {
+    this.moveActiveItemToNearestPaneInDirection('right', params)
+  }
+
+  moveActiveItemToNearestPaneInDirection (direction, params) {
+    const activePane = this.model.getActivePane()
+    const nearestPaneView = this.nearestVisiblePaneInDirection(direction, activePane)
+    if (nearestPaneView == null) { return }
+    if (params && params.keepOriginal) {
+      activePane.getContainer().copyActiveItemToPane(nearestPaneView.getModel())
+    } else {
+      activePane.getContainer().moveActiveItemToPane(nearestPaneView.getModel())
+    }
+    nearestPaneView.focus()
+  }
+
+  nearestVisiblePaneInDirection (direction, pane) {
+    const distance = function (pointA, pointB) {
+      const x = pointB.x - pointA.x
+      const y = pointB.y - pointA.y
+      return Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2))
+    }
 
-  moveActiveItemToPaneOnRight (params) { this.paneContainer.moveActiveItemToPaneOnRight(params) }
+    const paneView = pane.getElement()
+    const box = this.boundingBoxForPaneView(paneView)
+
+    const visiblePaneContainers = atom.workspace.getPaneContainers()
+      .filter(container => {
+        const isCenter = container === atom.workspace.getCenter()
```

Comment:
Again you could filter this down using shorter syntax;

```javascript
.filter(container => container === atom.workspace.getCenter || container.isVisible())
```

Additionally it might be worth putting the atom.workspace.getCenter() before the filter so it only gets evaluated once as it doesn't change per item.  Probably not much of a perf issue here but can be a problem inside larger collections or if the calls are slow.

```javascript
const center = atom.workspace.getCenter()
const visiblePaneContainers = atom.workspace.getPaneContainers)
   .filter(container => container === center || container.isVisible())
```
```